File: 22_09/22_09_05/pro-Immigration.py
def solution(n, times):
    left = 0
    right = max(times) *n
    answer = 0
    
    while left <= right:
        mid = (left+ right) //2
        checked = 0
        for time in times:
            checked += mid // time

            if checked >= n:
                break
        if checked >= n:
            answer = mid
            right = mid-1
        elif checked < n:
            left = mid+1
        
            
    return answer

File: 22_09/22_09_05/pro-immigration.js
function solution(n, times) {
    let answer = 0;
    let left = Math.min(...times)
    let right = Math.max(...times) * n

    while (left <= right){
        let mid = Math.floor((left + right)/2)
        let check_cnt = 0

        for (let i=0; i<times.length; i++){
            let time = times[i]
            check_cnt += Math.floor(mid/time);
            if (check_cnt >= n) break;
        }
        if (check_cnt >= n){
            answer = mid
            right = mid - 1
        }else {
            left = mid+1
        }
    }
    return answer;
}

console.log(solution(6, [7,10]))